lib/internal/util/comparisons.js

```diff
@@ -235,9 +235,23 @@ function innerDeepEqual(val1, val2, strict, memos) {
   } else if (isNativeError(val1) || val1 instanceof Error) {
     // Do not compare the stack as it might differ even though the error itself
     // is otherwise identical.
-    if ((!isNativeError(val2) && !(val2 instanceof Error)) ||
-        val1.message !== val2.message ||
-        val1.name !== val2.name) {
+    if (!isNativeError(val2) && !(val2 instanceof Error)) {
+      return false;
+    }
+
+    const message1Enumerable = ObjectPrototypePropertyIsEnumerable(val1, 'message');
+    const name1Enumerable = ObjectPrototypePropertyIsEnumerable(val1, 'name');
+    const cause1Enumerable = ObjectPrototypePropertyIsEnumerable(val1, 'cause');
+    const errors1Enumerable = ObjectPrototypePropertyIsEnumerable(val1, 'errors');
+
+    if ((message1Enumerable !== ObjectPrototypePropertyIsEnumerable(val2, 'message') ||
+          (!message1Enumerable && val1.message !== val2.message)) ||
+        (name1Enumerable !== ObjectPrototypePropertyIsEnumerable(val2, 'name') ||
+          (!name1Enumerable && val1.name !== val2.name)) ||
```

Comment:
As far as I can tell, this means now we **only** compare the `name` and `message` fields when they **are** enumerable, which specificaly opposite from the documentation, and is a breaking change...

Reply:
> As far as I can tell, this means now we **only** compare the `name` and `message` fields when they **are** enumerable, which specificaly opposite from the documentation, and is a breaking change...

FWIK, As mentioned in [ECMAScript 2025](https://tc39.es/ecma262/multipage/global-object.html#sec-constructor-properties-of-the-global-object-error), `name`, `message`, `cause` and `errors` are all  non-enumerable, so in the judgement above, I just compared the attributes whether they have same `enumerable` property, if it's true and they are `non-enumerable`, then compare the attributes contents. 

So if the attributes are `enumerable`(for example in a `custom Error`), the next check `keyCheck` will check them, in the doc change, I have just added  `errors`, and `causes` attributes, and didn't change `name` and `message`, I think its doesn't break the doc, have I missed something?

Reply:
If it was a breaking change, I would expect we would see test failures.